src/main/kotlin/anttracker/PageOf.kt

```diff
@@ -0,0 +1,119 @@
+package anttracker
+import org.jetbrains.exposed.sql.SizedIterable
+import org.jetbrains.exposed.sql.transactions.transaction
+import kotlin.math.ceil
+
+// A PageOf objects contains a max-20 element list of T.
+// Provides top-level functionality for:
+//  - loading DB records into memory (one page at a time)
+//  - accessing records
+//  - displaying records as a page
+//  - going to the next page and loading its records
+abstract class PageOf<IntEntity> {
+    protected var records: MutableList<IntEntity> = mutableListOf() // Container for object instances
+    protected var pagenum: Int = 0 // NOTE: Page numbers are 0-indexed
+    private var lastPageNum: Int = 0
+    protected val queryOffset: Int = 0 // For page calculation in DB query
+    protected val queryLimit: Int = 20 // Number of records per page.
+
+    // -------------------------------------------------------------------------------
+    // Prints to console a line-by-line display of objects contained in the PageOf instance.
+    // How an individual record is printed will be defined by subclass.
+    // If there are more pages to go, will indicate to user the no. of records to be seen.
+    // ---
+    fun display() {
+        for (record in records) {
+            printRecord(record)
+        }
+        if (!lastPage()) {
+            println("<Enter> for ${countRemainingRecords()} more.")
+        }
+    }
+
+    // -------------------------------------------------------------------------------
+    // Queries to DB and pulls a page of records (max 20) into the records MutableList.
+    // ---
+    fun loadRecords() {
```

Comment:
I believe we could have as the body as 

```kotlin
records = requireNotNull(queryToDB()).toList()
```

Or if `queryToDB` does not return `null`, then you can omit the `requireNotNull`.

Also, since you are changing the list every time, there is not need to make it mutable. Like this.

```kotlin
    private var records: List<T> = emptyList() // Container for object instances
```

Reply:
You are correct on all points. Fixed